components/mailgun/sources/new-bounce/new-bounce.js

```diff
@@ -0,0 +1,50 @@
+const common = require("../common-webhook");
+const { mailgun } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-bounce",
+  name: "New bounce",
+  description:
+    "Emit an event when the email recipient who could not be reached.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    domain: { propDefinition: [mailgun, "domain"] },
+  },
+  methods: {
+    ...common.methods,
+    getEventName() {
+      return ["bounce"];
+    },
+    getEventType() {
+      return "bounced";
+    },
+    generateMeta(eventPayload) {
+      const ts = eventPayload.timestamp;
+      return {
+        id: `${eventPayload["X-Mailgun-Sid"]}${ts}`,
+        summary: `New Bounce on message id: ${eventPayload["Message-Id"]} by ${eventPayload.recipient}`,
+        ts,
+      };
+    },
+    emitEvent(eventPayload) {
+      const eventTypes = this.getEventType();
+      if (eventTypes.includes(eventPayload.event)) {
+        const meta = this.generateMeta(eventPayload);
+        this.$emit(eventPayload, meta);
+      }
+    },
+  },
+  async run(eventRawData) {
+    const eventWorkload = eventRawData.body;
```

Comment:
This looks almost the same as the `run` method from the `common-webhook` file. Is there a reason it's unique?

Reply:
The reason why it is unique is that `maigun-new-bounce` is using a deprecated endpoint from Mailgun API that ultimately sends a different data structure when posting to the subscribed webhooks. In new Mailgun API endpoints `signature` and `event-data` are wrapped inside separated object, for `maigun-new-bounce` the signature and event data are inside the same object.

As an alternative we could have only one `run` method from `common-webhook` that asks for the `this.eventName` to see where to look for the signature and body.

In the Github issue for Maigun sources `maigun-new-bounce` this is included regardless of the deprecation status, and is implemented as well by the other services in the no-code database.

Reply:
Thanks for the explanation, having the separate method makes sense. Can you add this information (essentially the first paragraph of your explanation) as a comment in the code so that's clear?

Reply:
Ok, this information was added.